cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/environment/spark/SparkConfig.java

```diff
@@ -30,11 +31,14 @@ public class SparkConfig {
   private final String master;
   private final URI sparkJobFile;
   private final Map<String, String> configs;
+  private final CompletableFuture<Boolean> statusFuture;
 
-  public SparkConfig(String master, URI sparkJobFile, Map<String, String> configs) {
+  public SparkConfig(String master, URI sparkJobFile, Map<String, String> configs,
+                     CompletableFuture<Boolean> statusFuture) {
```

Comment:
Why there is a future in the config object?? Config shouldn't carry state.

Reply:
done